core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -186,7 +376,95 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public boolean doEquals(GeoBoundingBoxQueryBuilder other) {
+        return Objects.equals(topLeft, other.topLeft) &&
+                Objects.equals(bottomRight, other.bottomRight) &&
+                Objects.equals(type, other.type) &&
+                Objects.equals(normalize, other.normalize) &&
+                Objects.equals(fieldName, other.fieldName);
+    }
+
+    @Override
+    public int doHashCode() {
+        return Objects.hash(topLeft, bottomRight, type, normalize, fieldName);
+    }
+
+    @Override
+    public GeoBoundingBoxQueryBuilder doReadFrom(StreamInput in) throws IOException {
+        String fieldName = in.readString();
+        GeoBoundingBoxQueryBuilder geo = new GeoBoundingBoxQueryBuilder(fieldName);
+
+        double lat = in.readDouble();
+        double lon = in.readDouble();
+        geo.topLeft = new GeoPoint(lat, lon);
+
+        lat = in.readDouble();
+        lon = in.readDouble();
+        geo.bottomRight = new GeoPoint(lat, lon);
```

Comment:
super minor comment: re-using variables scares me a bit. I would prefer doing either new GeoPoint(in.readDouble(), in.readDouble()) which is actually less readable though, or use different variables, like topLeftLat, topLeftLon, bottomRightLat and bottomRightLon